src/app/services/user.service.ts

```diff
@@ -13,17 +16,33 @@ export class UserService {
 
   constructor(private http: HttpClient) { }
 
-  onCreateUser(user: any) {
+  // Uses backend api which creates user
+  onCreateUser(user: User) {
     return this.http.post<any>
-    ('https://mobile.openmf.org/fineract-provider/api/v1/users?username=mifos&password=password',
-     httpOptions);
+      ('https://mobile.openmf.org/fineract-provider/api/v1/users',
+        {
+          username: user.username,
+          firstname: user.firstname,
+          lastname: user.lastname,
+          email: user.email,
+          officeId: 1,
```

Comment:
@CajetanRodrigues hardcoded, this is what I meant by the previous comment. Without calling client API how you are getting the id of a client for further calling of APIs and for checking whether a user is a client or not.